protocols/kad/src/handler.rs

```diff
@@ -760,11 +760,14 @@ impl ConnectionHandler for Handler {
             }
 
             if self.outbound_substreams.len() < MAX_NUM_STREAMS {
-                if let Some((msg, id)) = self.pending_messages.pop_front() {
-                    self.queue_new_stream(id, msg);
-                    return Poll::Ready(ConnectionHandlerEvent::OutboundSubstreamRequest {
-                        protocol: SubstreamProtocol::new(self.protocol_config.clone(), ()),
-                    });
+                match self.pending_messages.poll_unpin(cx) {
+                    Poll::Ready((msg, id)) => {
+                        self.queue_new_stream(id, msg);
+                        return Poll::Ready(ConnectionHandlerEvent::OutboundSubstreamRequest {
+                            protocol: SubstreamProtocol::new(self.protocol_config.clone(), ()),
+                        });
+                    }
+                    Poll::Pending => {}
```

Comment:
Might want to use a if-let here.

```suggestion
                if let Poll::Ready((msg, id)) = self.pending_messages.poll_unpin(cx) {
                   self.queue_new_stream(id, msg);
                   return Poll::Ready(ConnectionHandlerEvent::OutboundSubstreamRequest {
                           protocol: SubstreamProtocol::new(self.protocol_config.clone(), ()),
                   });
                }
```

Reply:
Yes indeed. But I think that @thomaseizinger prefers that way to keep homogeneity. What do you think @thomaseizinger ?

Reply:
Don't really mind in this case. The rules I go by are usually:

1. Don't nest `match`, instead do as much pattern matching as possible in one go
2. If we are only interested in exactly one pattern, use `if let`
3. Avoid `else` as much as possible

So yeah, I'd probably use `if let` here (and did in the past looking at the diff :smiley: )

Reply:
Very well :wink: `if let` pattern done.